src/classes/CRLP_Batch_Base.cls

```diff
@@ -344,4 +359,45 @@ public abstract class CRLP_Batch_Base {
 
     }
 
+    /**
+     * @description returns the max Integer for all active time-bound Days_Back rollups,
+     * or null if there are no active time-bound Days_Back rollups for this summary object
+     */
+    private Integer maxDaysAgo = null;
+    protected Integer getMaxDaysAgo(){
+        if (maxDaysAgo == null) {
+            for (Rollup__mdt rollup : statefulCacheOfRollupsToBeProcessed) {
+                if (rollup.Active__c == true && rollup.Integer__c != null) {
```

Comment:
Is it worthy mooshing these three ifs into one? That's a big gnarly bunch of logic so not honestly sure if that's cleaner

Reply:
We're checking for maxDaysAgo == null on 368 and 373, so that could help? Generally I'd try to break complex logic like this into a separate method.